File: 2/duckinator.c
#include <stdio.h>

int main() {
  // Initial values
  unsigned int vals[2] = {1, 2};

  // When to stop
  int limit = 4000000; // 4,000,000

  // Return value
  unsigned int ret = 0;

  int i;
  unsigned int new;

  if ( vals[0] % 2 == 0 ) {
    ret += vals[0];
  }
  if ( vals[1] % 2 == 0 ) {
    ret += vals[1];
  }

  // Run (limit-2) times...
  for (i = 2; new <= limit; i++) {
    new = vals[0]+vals[1];
    vals[0] = vals[1];
    vals[1] = new;
    if (vals[1] % 2 == 0)
      ret += new;
  }

  printf("%u\n", ret); // Include the initial values!

  return 0;
}

File: 2/duckinator.rb
def duckinator
seed1, seed2 = 1, 2
limit = 4_000_000

i = seed2
ret = 0
new = 0

if seed1 % 2 == 0
  ret += seed1
end

if seed2 % 2 == 0
  ret += seed2
end

while seed2 <= limit
  new = seed1+seed2
  seed1, seed2 = seed2, new
  ret += new if seed2 % 2 == 0
end
puts ret

end
